Replace enterprise with license page

alxnddr opened the pull request:
Closes https://github.com/metabase/metabase/issues/18153

[[Product doc]](https://www.notion.so/metabase/Clean-up-settings-navigation-a85a0aff236d419c8681a03e6dbcaec2)

### Scope

#### 1) Enterprise page
We are removing the enterprise page which is available in the top navigation of the admin panel. Instead, on the settings page, we add the License section that shows different content based on the current subscription.

⚠️⚠️⚠️ When you switch from the OSS to EE editions in dev mode remove the Webpack cache using `yarn remove-webpack-cache`. Otherwise, it produces invalid states.

#### OSS
<img width="961" alt="Screenshot 2021-12-20 at 22 13 49" src="https://user-images.githubusercontent.com/14301985/146843568-0806c960-d451-4ba7-9196-c2717916a30d.png">

#### Starter
<img width="919" alt="Screenshot 2021-12-17 at 19 02 16" src="https://user-images.githubusercontent.com/14301985/146573046-13b32923-28dd-4840-a51d-2439feea8937.png">

#### Enterprise

**Unactivated**
<img width="1166" alt="unactivated enterprise" src="https://user-images.githubusercontent.com/14301985/147306201-b75fb2de-8ee4-47b6-b413-11e7ca648842.png">

**Trial**
<img width="1121" alt="trial enterprise" src="https://user-images.githubusercontent.com/14301985/147306214-73f9dc44-0c96-4054-b34e-ed6bd88b0bfb.png">

**Expired**
<img width="1097" alt="expired enterprise" src="https://user-images.githubusercontent.com/14301985/147306246-a01d4bb5-b81b-49b4-8634-c5441e1780c4.png">

**Cannot validate a token**
<img width="1190" alt="existing token network error" src="https://user-images.githubusercontent.com/14301985/147306270-2076758a-ec0f-4fb7-a867-6151d9adfcb9.png">

**Activated**
<img width="1092" alt="active enterprise" src="https://user-images.githubusercontent.com/14301985/147306287-ce6b38da-197a-4e0c-ad38-5b1c7585bb78.png">

#### 2) Premium embedding license page. 
We are deprecating the premium embedding plan, but for customers who already have the license key we added the new minimalistic premium embedding page. It is accessible under http://localhost:3000/admin/settings/premium-embedding-license url.

<img width="1187" alt="premium embedding not active" src="https://user-images.githubusercontent.com/14301985/147992270-ee681d0e-2942-4271-887e-d243e6669c0d.png">


### How to verify

#### License page

- Run Metabase in all configurations above (I cannot share tokens here) and ensure it looks as expected
- Run Metabase EE unactivated
    - Try to activate with an invalid token
    - Try to activate with a valid token
        - with `metabase-store-managed=true`
        - with `metabase-store-managed=false`

#### Premium embedding license page
- Run Metabase OSS
- Admin -> Embedding settings
- Enable embedding and scroll down to the top -> click `Activate it here` link
- Ensure the premium embedding license page is visible
- Ensure you are able to enter a valid token in order to activate the license

alxnddr committed:
remove old store page

alxnddr committed:
add license pages

alxnddr committed:
add remove webpack cache command

alxnddr committed:
lint

alxnddr committed:
fix specs

alxnddr committed:
remove license widget from the oss version

daltojohnso reviewed (approved):
looks good; just noted a missing translation

alxnddr committed:
add missing translation

alxnddr commented on enterprise/frontend/src/metabase-enterprise/license/components/LicenseWidget/LicenseWidget.tsx:
```diff
@@ -0,0 +1,67 @@
+
+  return (
+    <>
+      <SectionHeader>License</SectionHeader>
```
Good catch, thanks

alxnddr commented on enterprise/frontend/src/metabase-enterprise/license/components/LicenseAndBillingSettings/LicenseAndBillingSettings.tsx:
```diff
@@ -0,0 +1,113 @@
+      {!isActive && <UnlicensedContent />}
+
+      {isActive && (
+        <>
```
We've been able to use this notation since the Babel upgrade AFAIK

alxnddr committed:
show loader only for unput when updating token

alxnddr committed:
fix specs

flamber reviewed (commented):
This looks great.

Premium Embedding on OSS:
1. If someone is using the old Premium Embedding token, then it's not possible to input that in OSS. Would either need EE or using environment variable.
And even if I use the envvar, then the Admin panel doesn't give any indication of using it (neight on /admin/settings/embedding_in_other_applications nor /admin/settings/license), while it's actually working.
2. The link on OSS goes to /pricing/, whereas the "Explore paid features" basket goes to /upgrade. Shouldn't it match?
3. I cannot see if the license is active, the type of license, or when it will expire etc.

EE:
1. If I input a license, then it's not possible to change the license (except manually via in the appdb or using an envvar).
2. If I use an envvar, then the frontend still displays the active license, which is different from how envvars generally behave with other input fields.
3. I cannot see if the license is active, the type of license, which features it has, or when it will expire etc.
4. The last part of this sentence will cause confusion on the EE version: `Metabase is open source and will be free forever`
5. Isn't there a conflict between /admin/settings/license and /admin/settings/cloud, or is that just because the new token attributes are not available yet?
And shouldn't the license page for Cloud people let them know how they can get more functionality?
6. Can the token input field be a tiny bit wider, so it actually shows the full token input?
7. The placeholder includes dashes, like a Microsoft license, but our tokens doesn't have that.
8. If I input something incorrect (like `123`, or add dashes to my token, or copy from an email client that added a space before/after the token), then I have no idea how to Activate, since it's just disabled with no hints at all.

I currently cannot test an expired token, so I cannot tell how this behaves.

alxnddr committed:
amend the license page

alxnddr commented:
> 1. If someone is using the old Premium Embedding token, then it's not possible to input that in OSS. Would either need EE or using environment variable.
>    And even if I use the envvar, then the Admin panel doesn't give any indication of using it (neight on /admin/settings/embedding_in_other_applications nor /admin/settings/license), while it's actually working.

Correct me if I'm wrong, we decided to keep the ability to specify the token using envvar so nothing needs to be changed.

> 2\. The link on OSS goes to /pricing/, whereas the "Explore paid features" basket goes to /upgrade. Shouldn't it match?

Changed it to `/upgrade`

> 3\. I cannot see if the license is active, the type of license, or when it will expire etc.

@brunobergher @agnes-pinhanelli do we want to indicate that for the deprecated premium embedding plan?

> 1. If I input a license, then it's not possible to change the license (except manually via in the appdb or using an envvar).

I enabled the input

> 2\. If I use an envvar, then the frontend still displays the active license, which is different from how envvars generally behave with other input fields.

With the old enterprise page I believe it shows the same page as the active enterprise without usage of envvars. Is that correct? @brunobergher do we want to change it?

> 3\. I cannot see if the license is active, the type of license, which features it has, or when it will expire etc.

We added the expiration date and a distinction between trial/non-trial. @brunobergher @agnes-pinhanelli do we want to show a feature set now?

> 4\. The last part of this sentence will cause confusion on the EE version: `Metabase is open source and will be free forever`

We've removed that

> 5\. Isn't there a conflict between /admin/settings/license and /admin/settings/cloud, or is that just because the new token attributes are not available yet?
>     And shouldn't the license page for Cloud people let them know how they can get more functionality?

Hmm, it does look excessive. @brunobergher @agnes-pinhanelli should we change the license page or maybe just hide it for the starter plan?

> 6\. Can the token input field be a tiny bit wider, so it actually shows the full token input?
> 7\. The placeholder includes dashes, like a Microsoft license, but our tokens doesn't have that.
> 8\. If I input something incorrect (like `123`, or add dashes to my token, or copy from an email client that added a space before/after the token), then I have no idea how to Activate, since it's just disabled with no hints at all.

Fixed that.

@flamber thanks a lot for raising these points, I did not have enough knowledge in this area

alxnddr committed:
unactivated state fix

agnes-pinhanelli commented:
> @brunobergher @agnes-pinhanelli do we want to indicate that for the deprecated premium embedding plan?
> I don't think we need to, just mention that this function will be deprecated as we are doing today.
> We added the expiration date and a distinction between trial/non-trial. @brunobergher @agnes-pinhanelli do we want to show a feature set now?
>  I don't think we have to, this is just a transition state, we won't have the premium embedding anymore anyway, I think we can just mention that this is deprecated as shown on Figma.
> Hmm, it does look excessive. @brunobergher @agnes-pinhanelli should we change the license page or maybe just hide it for the starter plan?
> It does look excessive, I don't think we need to focus too much on this.

agnes-pinhanelli commented:
> @brunobergher @agnes-pinhanelli do we want to indicate that for the deprecated premium embedding plan?


> @alxnddr Nope


> We added the expiration date and a distinction between trial/non-trial. @brunobergher @agnes-pinhanelli do we want to show a feature set now?


> @alxnddr I don't think we need to

> Hmm, it does look excessive. @brunobergher @agnes-pinhanelli should we change the license page or maybe just hide it for the starter plan?


@alxnddr @brunobergher I don't think we need to focus too much on this but not sure if I'm following, [because we do mention more functionality in there.](https://www.figma.com/file/A5ouuvqtMkNXy78wrvuicQ/Clean-up-settings-navigation?node-id=58%3A4179)

alxnddr committed:
add premium embedding page, refactor, fix specs

alxnddr committed:
amend the license page

alxnddr committed:
amend the license page

nemanjaglumac commented:
Nit:
padding or margin for billing page is different from every other setting.

![image](https://user-images.githubusercontent.com/31325167/148094760-e49ad02d-8456-49ca-81af-ec6e002af141.png)

vs. 

![image](https://user-images.githubusercontent.com/31325167/148094823-1d001b9a-3c80-4ced-ac74-69c586643330.png)

Note how the text "jumps" for license page compared to other settings. Both left and top padding/margin are different.
![ ](https://p58.f1.n0.cdn.getcloudapp.com/items/wbuYJqyY/481780d6-4997-44da-bb9a-83a834c2e883.gif?source=viewer&v=2fd729f171442f7ee415bc7c92525033)

nemanjaglumac commented:
"Activate" button is active (pun intended :)) when the field is empty and it's possible to send a PUT request just like that. Shouldn't we disable it until someone enters anything in the input field?

Also, do we plan to do any FE validation for the token?

alxnddr commented:
> "Activate" button is active (pun intended :)) when the field is empty and it's possible to send a PUT request just like that. Shouldn't we disable it until someone enters anything in the input field?
> 
> Also, do we plan to do any FE validation for the token?

@flamber added a good point that it is not clear if we disable the button without any message. Since it is not a form and just one input with a button there won't likely be a blur event for validation so the server-side validation only looks legit. Regarding the empty field, it was possible previously to reset the token by sending the empty field so I kept this. Do you think we need to change that?

nemanjaglumac commented:
> Regarding the empty field, it was possible previously to reset the token by sending the empty field so I kept this. Do you think we need to change that?

According to the offline feedback I received, that functionality is intentionally there, so definitely leave it. :)

alxnddr committed:
change license page padding
